samza-sql/src/main/java/org/apache/samza/sql/SamzaSqlInputTransformer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.sql;
+
+import java.time.Instant;
+import org.apache.samza.operators.KV;
+import org.apache.samza.sql.data.SamzaSqlRelMsgMetadata;
+import org.apache.samza.system.IncomingMessageEnvelope;
+import org.apache.samza.system.descriptors.InputTransformer;
+
+
+/**
+ * SamzaSqlInputTransformer:
+ *   This transformer retrieves the message from the {@link IncomingMessageEnvelope}, and if it is not an
+ *   Instance of KV, it creates KV. In both cases the KV is wrapped into a {@link MessageAndMetadata}
+ *   object, to copy over the message's meatadata (eventTime and arrivalTime)
+ */
+
+public class SamzaSqlInputTransformer implements InputTransformer {
+
+  @Override
+  public Object apply(IncomingMessageEnvelope ime) {
+    Object samzaMsg;
+    if (ime.getMessage() instanceof KV) {
```

Comment:
message is always not KV (I ran experiment and verify that). We get here from InputOperatorImpl when transformer is not null. Before introducing the SamzaSQLInputTransformer, InputOperatorImpl had transformer is null & isKeyed is true (because noOpKVSerde is KVSerde), and would  always created a KV. Now, with transformer is not null, the output is whatever we generate, so to have same effect we need to create the KV. I changed the code to always create the KV.